train.py

```diff
@@ -75,6 +78,8 @@ def setup_loader(is_val=False, verbose=False):
 def train(model, criterion, criterion_st, optimizer, optimizer_st, scheduler,
           ap, epoch):
     data_loader = setup_loader(is_val=False, verbose=(epoch==0))
+    if c.num_speakers > 0:
```

Comment:
Why don't we compute num_speakers on the fly?

Reply:
That would also deprecate the line 105

Reply:
because you would need to iterate through all datasets before knowing the number of speakers

Reply:
I don't think another config var is necessary. If num_spearkers is 1 we can turn it off and on otherwise. 

In theory, each dataset processor should parse the whole dataset for keeping the same standard between the runs. Therefore, I'd say, it is sound to assume we have all the speakers loaded in advance. 

Merging >two datasets should be also easy to list them in the config file. We can set a list of target datasets in ```dataset``` field and merge all the speakers they have. 

If you need to use a sub-set of a dataset, you can create a symbolic link to target files in another folder and use the same dataloader.

Hope makes sense

Reply:
Can you clarify what you mean by **keeping the same standard** between runs? We currently keep the same mapping between runs, and it is possible to also add and remove speakers in a subsequent run or finetuning

Reply:
Maybe one thing that would be desirable is that you keep the same mapping inbetween independent training sessions. Is that what you mean?

Reply:
yes this is what I mean

Reply:
If you could have a look that be great, did some basic testing but didn't manage to do so extensively yet. Might have more time later today or tomorrow